core/src/main/java/org/elasticsearch/gateway/Gateway.java

```diff
@@ -69,7 +68,7 @@ public void performStateRecovery(final GatewayStateRecoveredListener listener) t
         TransportNodesListGatewayMetaState.NodesGatewayMetaState nodesState = listGatewayMetaState.list(nodesIds, null).actionGet();
 
 
-        int requiredAllocation = Math.max(1, minimumMasterNodesProvider.get());
+        int requiredAllocation = Math.max(1, minimumMasterNodes);
```

Comment:
shall we push this to the constructor and make `requiredAllocations` a final field instead of `minmumMasterNodes`?